interfaces/builtin/media_hub.go

```diff
@@ -0,0 +1,200 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"bytes"
+
+	"github.com/snapcore/snapd/interfaces"
+)
+
+const mediaHubPermanentSlotAppArmor = `
+# Description: Allow operating as the the media-hub service.
+
+# DBus accesses
+#include <abstractions/dbus-session-strict>
+
+dbus (send)
+    bus=session
+    path=/org/freedesktop/DBus
+    interface=org.freedesktop.DBus
+    member="{Request,Release}Name"
+    peer=(name=org.freedesktop.DBus, label=unconfined),
+
+# Allow binding the service to the requested connection name
+dbus (bind)
+    bus=session
+    name="core.ubuntu.media.Service",
+
+dbus (receive, send)
+    bus=session
+    path=/com/ubuntu/media/Service{,/**}
+    interface=org.freedesktop.DBus{,.*}
+    peer=(label=unconfined),
+
+dbus (receive)
+    bus=session
+    interface=org.freedesktop.DBus.Introspectable
+    peer=(label=unconfined),
+
+dbus (receive, send)
+    bus=session
+    path=/core/ubuntu/media/Service{,/**}
+    peer=(label=unconfined),
+`
+
+const mediaHubConnectedSlotAppArmor = `
+# Allow connected clients to interact with the player
+dbus (receive)
+    bus=session
+    interface=org.freedesktop.DBus.Properties
+    path=/core/ubuntu/media/Service
+    peer=(label=###PLUG_SECURITY_TAGS###),
+
+dbus (receive)
+    bus=session
+    interface=org.freedesktop.DBus.Introspectable
+    peer=(label=###PLUG_SECURITY_TAGS###),
+
+dbus (receive)
+    bus=session
+    interface="core.ubuntu.media.Service{,.*}"
+    path=/core/ubuntu/media/Service
+    peer=(label=###PLUG_SECURITY_TAGS###),
+
+# Allow clients to manage Player sessions
+dbus (receive)
+    bus=session
+    path=/core/ubuntu/media/Service
+    interface=core.ubuntu.media.Service
+    member="{Create,Detach,Reattach,Destroy,CreateFixed,Resume}Session"
+    peer=(label=###PLUG_SECURITY_TAGS###),
+
+# Allow clients to pause all other sessions
+dbus (receive)
+    bus=session
+    path=/core/ubuntu/media/Service
+    interface=core.ubuntu.media.Service
+    member="PauseOtherSessions"
+    peer=(label=###PLUG_SECURITY_TAGS###),
+
+# Allow clients to query/modify service properties
+dbus (receive)
+    bus=session
+    path=/core/ubuntu/media/Service{,/**}
+    peer=(label=###PLUG_SECURITY_TAGS###),
```

Comment:
This is a superset of every receive rule that uses this path. Can you add a path= or member= clause to this rule?